apps/web/test/web/controllers/github_webhook_controller_test.exs

```diff
@@ -25,22 +23,51 @@ defmodule Web.GitHubWebhookControllerTest do
         "closed_at": nil
       },
       "repository": %{
-        "name": "public-repo",
-        "html_url": "https://.github.com/repos/elixirschool/extracurricular"
+        "name": project.name,
+        "html_url": project.url,
       },
       "sender": %{
         "login": "doomspork"
       }
     }
+  end
+
+  describe "with existing project" do
+    setup do
+      {:ok, project} = Projects.insert(%{name: "public-repo", url: "https://github.com/repos/elixirschool/extracurricular"})
+      [project: project]
+    end
+    test "POST /webhooks/github creates opportunity", %{conn: conn, project: project} do
+      # Fields omitted from official webhook payload for brevity
+
+      conn =
+        conn
+        |> put_req_header("content-type", "application/json")
+        |> post("/webhooks/github", Poison.encode!(webhook_body_for_project(project)))
```

Comment:
We'd rather not nest functions: https://github.com/christopheradams/elixir_style_guide#pipe-operator

How about something like:

```elixir
payload =
  project
  |> webhook_body_for_project
  |> Poison.encode!

conn =
 conn
 |> put_req_header("content-type", "application/json")
 |> post("/webhooks/github", payload)
```

Reply:
That's what I get for working in Rust recently. :facepalm:

Since the payload encoding isn't specifically germane to what those tests are validating, I moved it out into the setup context.

Reply:
Rust?!  You're hardcore 😁